mdx-web/src/test/groovy/com/mx/path/model/mdx/web/controller/AccountNumbersControllerTest.groovy

```diff
@@ -35,18 +35,34 @@ class AccountNumbersControllerTest extends Specification {
     RequestContext.clear()
   }
 
-  def "gets account numbers for an account"() {
+  def "gets on demand account numbers for an account"() {
     given:
     AccountsController.setGateway(gateway)
     def accountNumbers = new AccountNumbers()
 
     when:
-    doReturn(new AccessorResponse<AccountNumbers>().withResult(accountNumbers)).when(accountNumberGateway).get("A-123")
-    def result = subject.get("A-123")
+    doReturn(new AccessorResponse<AccountNumbers>().withResult(accountNumbers)).when(accountNumberGateway).get("A-123-OD")
+    def result = subject.getOnDemandAccountNumbers("A-123-OD")
 
     then:
     verify(accountGateway).accountNumbers() || true
-    verify(accountNumberGateway).get("A-123") || true
+    verify(accountNumberGateway).get("A-123-OD") || true
+    result.getBody() == accountNumbers
+    RequestContext.current().feature == "accounts"
+  }
+
+  def "gets mdx account numbers for an account"() {
+    given:
+    AccountsController.setGateway(gateway)
+    def accountNumbers = new AccountNumbers()
+
+    when:
+    doReturn(new AccessorResponse<AccountNumbers>().withResult(accountNumbers)).when(accountNumberGateway).get("A-123-MDX")
+    def result = subject.getAccountNumbers("A-123-MDX")
+
+    then:
+    verify(accountGateway).accountNumbers() || true
+    verify(accountNumberGateway).get("A-123-MDX") || true
     result.getBody() == accountNumbers
     RequestContext.current().feature == "accounts"
```

Comment:
`RequestContext.current().feature == "accounts"` can be removed if we aren't calling `ensureFeature` anymore.

Reply:
Gotcha, I'll take it out